src/server/pfs/pretty/pretty.go

```diff
@@ -34,26 +34,34 @@ func PrintRepoHeader(w io.Writer, printAuth bool) {
 }
 
 // PrintRepoInfo pretty-prints repo info.
-func PrintRepoInfo(w io.Writer, repoInfo *pfs.RepoInfo) {
+func PrintRepoInfo(w io.Writer, repoInfo *pfs.RepoInfo, fullTimestamp bool) {
 	fmt.Fprintf(w, "%s\t", repoInfo.Repo.Name)
-	fmt.Fprintf(
-		w,
-		"%s\t",
-		pretty.Ago(repoInfo.Created),
-	)
+	if fullTimestamp {
+		fmt.Fprintf(w, "%s\t", repoInfo.Created.String())
+	} else {
+		fmt.Fprintf(w, "%s\t", pretty.Ago(repoInfo.Created))
+	}
 	fmt.Fprintf(w, "%s\t", units.BytesSize(float64(repoInfo.SizeBytes)))
 	if repoInfo.AuthInfo != nil {
 		fmt.Fprintf(w, "%s\t", repoInfo.AuthInfo.AccessLevel.String())
 	}
 	fmt.Fprintln(w)
 }
 
+// PrintableRepoInfo is a wrapper around RepoInfo containing any formatting options
+// used within the template to conditionally print information.
+type PrintableRepoInfo struct {
+	*pfs.RepoInfo
+	FullTimestamp bool
+}
+
 // PrintDetailedRepoInfo pretty-prints detailed repo info.
-func PrintDetailedRepoInfo(repoInfo *pfs.RepoInfo) error {
+func PrintDetailedRepoInfo(repoInfo *PrintableRepoInfo) error {
```

Comment:
@gabrielgrant we needed to find a way to pass the flag information to these `PrintDetailed*` functions. Correct me if I'm wrong, but to the best of my knowledge, there is no way to pass extra info into go templates.

The solution here is to wrap the struct we are generating our template off with the flag data. Another option is to just convert all these `PrintDetailed*` functions into separate `Fprintf` statements instead of using templates, but that seemed a bit more unwieldy. Thoughts?

Reply:
wrapping the struct seems reasonable to me, but I'm far from the expert on these sorts of stylistic Go issues. Probably best to ask @jdoliner (or another systems team member) if you're really concerned